docs/pages/Cardinal/CoreConcepts/Components.mdx

```diff
@@ -0,0 +1,109 @@
+# Components
+
+Components are the structures that give entities their state.
+
+## Creating a Component
+
+In Cardinal, you define your component using a normal Go struct.
+
+```go
+type Health struct {
+    Amount  uint64
+    Cap     uint64
+}
+```
+
+In order to use this struct in the ECS world, instantiate a new component type, passing your struct in the type
+parameter.
+
+```go
+import "github.com/argus-labs/world-engine/cardinal/ecs"
+
+var HealthComponent = ecs.NewComponentType[Health]()
+
+```
+
+## Getting Component Data
+Component data is retrieved via the `Get` method of the corresponding component type. `Get` requires a pointer to `World` and the ID of the entity with this component, and returns the component struct and an error.
+
+```go
+health, err := HealthComponent.Get(world, id)
+```
+
+## Setting Component Data
+Components are updated via the `Set` method of the corresponding component type. `Set` requires a pointer `World`, the ID of the entity with this component, and the go struct that defines this component. `Set` may return an error.
+
+```go
+err := HealthComponent.Set(world, id, Health{Amount: 50, Cap: 100})
+```
+
+## Updating Component Data
+Aside from directly setting component data (effectively a full overwrite), component types provide a method `Update`, which updates component data with a more ergonomic callback pattern. `Update` takes a pointer to `World`, the ID of the entity to update, and a callback function. The
+callback function supplies the component struct directly, and expects a return value of the updated component struct. `Update` may return an error.
+
+```
+err := HealthComponent.Update(world, id, func(h Health) Health {
+    // take away 10 from the total heatlh
+    h.Amount -= 10
+    return h
+})
+```
```

Comment:
Optional: What do you think about including something like:

The `Update` pattern above is equivalent to:

```go
    h, err := HealthComponent.Get(world, id)
    if err != nil {
        return err
    }
    h.Amount -= 10
    err = HealthComponent.Set(world, id, h)
    if err != nil {
        return err
    }
```